### ISSUE TITLE
Can't use `AvroBaseModel` and `faust.Record` together...

### ISSUE THREAD

**Issue by rightx2** [marcosschroh/dataclasses-avroschema#207]
**Version**
dataclasses-avroschema==0.32.1

**Describe the bug**
1. Can't use `AvroBaseModel` and `faust.Record` together!
2. default field is ignored...

**To Reproduce**
```
import faust

import enum
import dataclasses

from dataclasses_avroschema.avrodantic import AvroBaseModel


class FavoriteColor(str, enum.Enum):
    BLUE = "BLUE"
    YELLOW = "YELLOW"
    GREEN = "GREEN"


@dataclasses.dataclass
class UserAdvance(faust.Record, AvroBaseModel):
    name: str
    age: int
    country: str = "Argentina"
    address: str = None

    class Meta:
        schema_doc = False

# problem 1. raise ValueError
user = UserAdvance(name="bond", age="Asdf", country="asdf", address="Asdf")
>> ValueError: "UserAdvance" object has no field "__evaluated_fields__"


# problem2. default field ignored
user = UserAdvance(name="bond", age="Asdf")
>> TypeError: __init__() missing 2 required positional arguments: 'country' and 'address'

```

**Comment by marcosschroh** [comment]
Hi @rightx2

You can not use `faust.Records` and `AvroBaseModel` (extension of `pydantic` ) at the same time. The reason is that `pydantic` and faust` have their own way of creating and validating class instances. Both of them have a `__new__` method that introduces their "magic" to validate the data. 

The following example with `faust` and `pydantic` (`BaseModel` instead of `AvroBaseModel`) also does not work:

```python
import faust
from pydantic import BaseModel


class UserAdvance(faust.Record, BaseModel):
    name: str
    age: int
    country: str = "Argentina"
    address: str = None

    class Meta:
        schema_doc = False

user = UserAdvance(name="bond", age="Asdf", country="asdf", address="Asdf")
>> ValueError: "UserAdvance" object has no field "__evaluated_fields__"
```

In summary, you can use:

- [X] `faust.Record` with `AvroModel`
- [X] `pydantic.BaseModel` with `AvroModel`  = `AvroBaseModel`
